core/src/main/java/kafka/server/SharePartition.java

```diff
@@ -42,6 +44,7 @@
 import org.apache.kafka.storage.internals.log.FetchPartitionData;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import scala.Option;
```

Comment:
I would really strongly suggest not having a dependency on this if we can avoid it. Maybe this needs a small change to the replica manager.

Reply:
Though I can see this being added in other java classes a well but good to remove if possible. But maybe replica manager refactoring might be heavy on other classes